### PR TITLE
Add new policy for "hsqldb". #1241456

### PR DESCRIPTION
**Pull request by vmojzis**

### CODE REVIEW

**Comment by mgrepl** [comment]
So if it is needed by default, it should be called outside of auth_use_nsswitch().

File: hsqldb.if [hunk 1]
```diff
@@ -0,0 +1,247 @@
+
+## <summary>policy for hsqldb</summary>
```
Comment (wrabcak) [inline hsqldb.if]: Here please add some summary of this daemon.

File: hsqldb.if [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+## <summary>
+##	All of the rules required to administrate
+##	an hsqldb environment
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+## <param name="role">
```
Comment (wrabcak) [inline hsqldb.if]: Too many parameters for this inteface.

File: hsqldb.te [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+allow hsqldb_t self:process execmem;
+
+allow hsqldb_t self:fifo_file rw_fifo_file_perms;
+allow hsqldb_t self:stream_socket_class_set create_stream_socket_perms;
+
+manage_dirs_pattern(hsqldb_t, hsqldb_tmp_t, hsqldb_tmp_t)
+manage_files_pattern(hsqldb_t, hsqldb_tmp_t, hsqldb_tmp_t)
+manage_lnk_files_pattern(hsqldb_t, hsqldb_tmp_t, hsqldb_tmp_t)
+files_tmp_filetrans(hsqldb_t, hsqldb_tmp_t, { dir file lnk_file })
```
Comment (wrabcak) [inline hsqldb.te]: Do we need here also file and lnk_file?

File: hsqldb.te [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+corenet_tcp_bind_tor_port(hsqldb_t)
+corenet_tcp_connect_tor_port(hsqldb_t)
+
+dev_list_sysfs(hsqldb_t)
+
+dev_read_urand(hsqldb_t)
+dev_read_rand(hsqldb_t)
+
+
+auth_use_nsswitch(hsqldb_t)
```
Comment (wrabcak) [inline hsqldb.te]: Follow alphabetical order please ;)

File: hsqldb.te [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+type hsqldb_unit_file_t;
+systemd_unit_file(hsqldb_unit_file_t)
+
+########################################
+#
+# hsqldb local policy 
+# 
+
+allow hsqldb_t self:process execmem;
```
Comment (wrabcak) [inline hsqldb.te]: Could you start this daemon without this rule?

File: hsqldb.te (hunk 4 above)
Comment (doverride) [inline hsqldb.te]: On Thu, Aug 13, 2015 at 05:40:48AM -0700, Lukas Vrabec wrote:
> > +kernel_read_system_state(hsqldb_t)
> > +kernel_read_network_state(hsqldb_t)
> > +
> > +corecmd_exec_bin(hsqldb_t)
> > +
> > +corenet_tcp_bind_generic_node(hsqldb_t)
> > +corenet_tcp_bind_tor_port(hsqldb_t)
> > +corenet_tcp_connect_tor_port(hsqldb_t)
> > +
> > +dev_list_sysfs(hsqldb_t)
> > +
> > +dev_read_urand(hsqldb_t)
> > +dev_read_rand(hsqldb_t)
> > +
> > +
> > +auth_use_nsswitch(hsqldb_t)
> 
> Follow alphabetical order please ;) 

The snippet above is actually close to proper.

The auth module is not part of the kernel layer and should be below any kernel layer interface calls as per:

https://github.com/TresysTechnology/refpolicy/wiki/StyleGuide

In particular:

1. calls to kernel layer modules, sorted in alphabetical order (exception: kernel module is first)
2. calls to system layer modules, sorted in alphabetical order
3. calls to all other modules, sorted in alphabetical order


> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/fedora-selinux/selinux-policy/pull/30/files#r36969159

-- 
02DFF788
4D30 903A 1CF3 B756 FB48  1514 3148 83A2 02DF F788
http://keys.gnupg.net/pks/lookup?op=vindex&search=0x314883A202DFF788
Dominick Grift

File: hsqldb.te (hunk 3 above)
Comment (vmojzis) [inline hsqldb.te]: Just "dir" and "file", thanks.

File: hsqldb.te (hunk 5 above)
Comment (vmojzis) [inline hsqldb.te]: No, java needs it.

**Pull request merged**